### PR TITLE
Usability updates

### PR DESCRIPTION
**Pull request by chelnak**
This PR includes:

* A new makefile that offers a consistent way to provision boxes
* Proper configuration for system gems/bundler
* Consistency updates for docs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+.DEFAULT_GOAL := help
+
+log_error = (>&2 echo "\x1B[31m>> Error:$1\x1B[39m\n" && $(MAKE) help && exit 1)
+
+.PHONY: help
+help:
+	@echo "Usage: make [os] [version] [command]"
+	@echo "\n"
+	@echo "Examples:"
+	@echo "  make ubuntu 2004 up"
```
Comment (GSPatton) [inline Makefile:14]: there is no 2004 target, so `make ubuntu 2004 up` fails

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+server-2022:
+	$(eval VERSION=server-2022)
+
+# ============= Ubuntu ============= #
+.PHONY: ubuntu
+ubuntu:
+	$(eval PLAT=ubuntu)
+
+2204:
```
Comment (GSPatton) [inline Makefile:48]: there needs to be an `ubuntu/2204` working directory for this target to work. At the moment this only exists for `ubuntu/2004`

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+server-2012R2:
+	$(eval VERSION=server-2012R2)
+
+server-2016:
+	$(eval VERSION=server-2016)
+
+server-2019:
+	$(eval VERSION=server-2019)
+
+server-2022:
```
Comment (GSPatton) [inline Makefile:40]: unrelated to this PR but server-2022 relies on `windows/scripts/bootstrap.ps1` for provisioning which doesn't exist

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	bolt module install --project ./windows/bootstrap
+endif
+
+server-2012R2:
+	$(eval VERSION=server-2012R2)
+
+server-2016:
+	$(eval VERSION=server-2016)
+
+server-2019:
```
Comment (GSPatton) [inline Makefile:37]: Unrelated to this PR but `server-2019` now uses bolt provisioning which seems to confuse vagrant

`vm:
* The 'bolt' provisioner could not be found.

Vagrant:
* Unknown configuration section 'bolt'.`

The vagrant [docs](https://developer.hashicorp.com/vagrant/docs/provisioning/puppet_apply) don't have bolt listed as a known provisioner. Only `puppet apply` and `puppet agent`. 

Is this bolt provisioner a custom provisioner of some sort?

Comment (chelnak) [reply Makefile:37]: Yeah you need to install the bolt plugin for vagrant.. i thought i'd updated the readme

Comment (chelnak) [reply Makefile:37]: oh yes I did - its on the main readme under requirements.

**Review by GSPatton — APPROVED** [review]

**Pull request merged**